maestrowf/datastructures/yamlspecification.py

```diff
@@ -344,16 +346,30 @@ def verify_parameters(self):
                         raise ValueError("Missing {} keys in the global "
                                          "parameter named {}"
                                          .format(missing_attrs, name))
+                    # If label is a list, check its length against values.
+                    values = value["values"]
+                    label = value["label"]
+                    if isinstance(label, list):
+                        if len(values) != len(label):
+                            raise ValueError("Global parameter '{}' the "
+                                             "values length does not "
+                                             "match the label list length."
+                                             .format(name))
+                        if len(label) != len(set(label)):
+                            raise ValueError("Global parameter '{}' the "
+                                             "label does not contain "
+                                             "unique labels."
+                                             .format(name))
                     # Add the name to global parameters encountered, check if
                     # length of values is the same as previously encountered.
                     global_names.add(name)
                     # If length not set, set it and continue
-                    if value_len == -1:
-                        value_len = len(value)
+                    if values_len == -1:
+                        values_len = len(values)
                         continue
 
                     # Check length. Exception if doesn't match.
-                    if len(value) != value_len:
+                    if len(values) != values_len:
```

Comment:
Oh, you did miss a conditional here. You'll need to do the following:
```
if isinstance(value["label"], list):
    if len(value) != value_len:
        raise ValueError("Label list for global parameter '{}' is" 
                                   "not the as the list of its values."
                                    .format(name))
```

Reply:
Interestingly, I think there may be a bug here also. I'm looking into it currently.

Reply:
@FrankD412 I think that is included, if the values and labels lists are the same length and all of the global values lengths agree, then the the configuration is sound.

Reply:
@koning -- Oh sorry, I may have been looking at an older version in my sea of tabs... I see it now.

I did some verifying on my end, and your having me review this segment of my original code did expose that I do in fact have a bug here. It looks like you caught it and fixed it. Let me review it one more time to be safe.

Reply:
Looks like you fixed it -- thanks for correcting it!